packages/contracts/contracts/newsroom/Newsroom.sol

```diff
@@ -184,27 +191,44 @@ contract Newsroom is ACL {
     require(content.author == 0x0 || content.author == author);
     require(content.revisions.length > revisionId);
 
+    if (contentId == 0) {
+      require(isOwner(msg.sender));
+    }
+
     content.author = author;
 
     Revision storage revision = content.revisions[revisionId];
     revision.signature = signature;
 
-    require(verifyRevisionSignature(author, revision));
+    require(verifyRevisionSignature(author, contentId, revision));
 
     emit RevisionSigned(contentId, revisionId, author);
   }
 
-  function verifyRevisionSignature(address author, Revision storage revision) view internal returns (bool isSigned) {
+  function verifyRevisionSignature(address author, uint contentId, Revision storage revision) internal returns (bool isSigned) {
     if (author == 0x0 || revision.signature.length == 0) {
       require(revision.signature.length == 0);
       return false;
     } else {
+      // The url is is not used in the cryptography by design,
+      // the rationale is that the Author can approve the content and the Editor might need to set the url
+      // after the fact, or things like DNS change, meaning there would be a question of canonical url for that article
```

Comment:
I was under the impression that the contentUri is an IPFS hash. The comments indicate that this might change in the future allowing content from URLs?

Reply:
@thec00n The Uri is whatever that specific Newsroom wants it to be. We're making multiple URI solutions, with basics such as HTTP, IPFS and more experimental such as using Event Log of a Smart Contract or appended data at the end of the transaction

Reply:
For unsigned and signed content any Civil client needs to download the content at contentUri and generate a hash and verify that the hash matches the content that is stored at contentUri. It needs to be ensured by the Civil client that no invalid content is displayed to the user because by design a user could add a contentUri where the contentHash does not match its content. This is could be a nuisance and used to attack the Civil system.

Reply:
@thec00n I agree, this is a trade-off we have to do due to content being outside of Ethereum.

Reply:
Ok, just want to make sure that you guys are aware that this could be a problem.